wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/QR Code/Models/UserQRCodeViewModel.swift

```diff
@@ -27,10 +27,11 @@ final class UserQRCodeViewModel: ObservableObject {
 
     init(
         profileLink: String,
+        profileDeepLink: String,
         handle: String
     ) {
         self.profileLink = profileLink
-        let qrCodeImage = QRCodeGenerator.generateQRCode(from: profileLink)
+        let qrCodeImage = QRCodeGenerator.generateQRCode(from: profileDeepLink)
```

Comment:
What happens if someone scans the QR code and they don't have the app installed? A much better flow would be to transition to the App Store so they can download the app. I think it can be achieved by introducing a method like this one:

```swift
func openProfileDeepLink() {
        guard let appURL = URL(string: profileDeepLink),
              let appStoreURL = URL(string: "https://apps.apple.com/app/idYOUR_APP_ID") else { return }

        // Check if the app can open the deep link
        if UIApplication.shared.canOpenURL(appURL) {
            UIApplication.shared.open(appURL, options: [:], completionHandler: nil)
        } else {
            // Redirect to the App Store if the app is not installed
            UIApplication.shared.open(appStoreURL, options: [:], completionHandler: nil)
        }
    }
}
```

Then we can call this method through our viewModel wherever we want.